vision-quickstart (1)/app/src/main/java/com/google/mlkit/vision/demo/textdetector/TextRecognitionProcessor.java

```diff
@@ -62,6 +64,19 @@ public void stop() {
     @Override
     protected void onSuccess(@NonNull Text text, @NonNull GraphicOverlay graphicOverlay) {
         Log.d(TAG, "On-device Text detection successful");
+        recogonizedTextList.add(text.getTextBlocks().get(0).getLines().get(0).getText());
+        if(recogonizedTextList.size()==depth)
+        {
+            Log.d("Noise testing", "Frame id: " + 0 + "  Frame text:" + recogonizedTextList.get(0));
+            Integer differentFrames = 0;
+            for(int i=1;i<depth;i++) {
+                Log.d("Noise testing", "Frame id: " + i + "  Frame text:" + recogonizedTextList.get(i));
+                if(recogonizedTextList.get(i).equals(recogonizedTextList.get(i-1)) == false)
```

Comment:
stylistically, == false is a bit unusual, normally it would be 
if(!recogonizedTextList.get(i).equals(recogonizedTextList.get(i-1))) {

Also, it is recommended to always use braces on if statements, even if it is a single line. https://engdoc.corp.google.com/eng/doc/devguide/java/style/index.md?cl=head#s4.1.1-braces-always-used

This is so if you add another statement you don't need to remember to add the braces, they are already there.  This is a common source of logic errors (maybe not so much these days with auto-indenting IDEs, but still...)

Reply:
Understood! Thanks for the pointers.